server/GameServer.cpp

```diff
@@ -359,69 +359,71 @@ void GameServer::ProcessRequestExternalAddress(Packet& packet, sockaddr_in& from
 	}
 }
 
-void GameServer::DoTimedUpdates()
+void GameServer::CheckTimedUpdates()
 {
-	#ifdef DEBUG
-		//LogMessage("DoTimedUpdates()");
-	#endif
-
-	// Get the current time
 	auto currentTime = std::time(nullptr);
+
 	// Check for timed out game entries
 	for (std::size_t i = gameSessions.size(); i-- > 0; )
 	{
-		// Get the current time difference
 		auto timeDiff = currentTime - gameSessions[i].time;
-
-		// Check for no initial update within required time
-		if ((timeDiff >= InitialReplyTime) && ((gameSessions[i].flags & GameSessionReceived) == 0))
-		{
-			LogEndpoint("Dropping Game: No initial Host Info from: ", gameSessions[i].addr.sin_addr.s_addr, gameSessions[i].addr.sin_port);
-
-			// Clear bogus entry
-			FreeGameSession(i);
-			// Update counters
-			counters.numDroppedHostedPokes++;
-		}
+		
 		// Check if no updates have occured for a while
-		else if ((timeDiff >= UpdateTime) && ((gameSessions[i].flags & GameSessionReceived) != 0))
+		if (timeDiff >= InitialReplyTime)
 		{
-			// Entry is old and requires update
-			// --------------------------------
+			DoTimedUpdate(i, timeDiff);
+		}
+	}
 
-			// Check if enough time has elapsed to give up
-			if (timeDiff >= GiveUpTime)
-			{
-				LogEndpoint("Dropping Game: Lost contact with host: ", gameSessions[i].addr.sin_addr.s_addr, gameSessions[i].addr.sin_port);
+	// Check if we should reduce memory usage **TODO**

```

Comment:
What is meant by this comment (perhaps we can wholesale delete it, or add an issue to address and then delete it)?